pkg/kubectl/cmd/run.go

```diff
@@ -117,9 +117,11 @@ func addRunFlags(cmd *cobra.Command) {
 	cmd.Flags().Bool("expose", false, "If true, a public, external service is created for the container(s) which are run")
 	cmd.Flags().String("service-generator", "service/v2", "The name of the generator to use for creating a service.  Only used if --expose is true")
 	cmd.Flags().String("service-overrides", "", "An inline JSON override for the generated service object. If this is non-empty, it is used to override the generated object. Requires that the object supply a valid apiVersion field.  Only used if --expose is true.")
+	cmd.Flags().Bool("silent", false, "If true, hide the 'waiting for pod to be running' message")
```

Comment:
I would worry that "--silent" here might be interpreted to mean "show no
log output, only return an exit code if you can get the pod"

On Mon, Jul 11, 2016 at 6:49 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Isn't this normally "quiet"?   I'm trying to search man on my system for a
> rough count of either, but it's taking too long.
>
> On Mon, Jul 11, 2016 at 6:43 PM, Clayton Coleman <ccoleman@redhat.com>
> wrote:
>
>> Maybe we should get this in one place first and then come back later and
>> make it global.  I'd like us to refactor the PrinterCommand stuff because
>> it's not aligned with how we do option structs (ideally we'd create an
>> option struct that we invoke bind on for the default printer options, and
>> hide retrieving the printer logic from commands).
>>
>> On Mon, Jul 11, 2016 at 6:35 PM, Jeff Lowdermilk <
>> notifications@github.com> wrote:
>>
>>> In pkg/kubectl/cmd/run.go
>>> <https://github.com/kubernetes/kubernetes/pull/28801#discussion_r70349834>
>>> :
>>>
>>> > @@ -117,9 +117,11 @@ func addRunFlags(cmd *cobra.Command) {
>>> >  	cmd.Flags().Bool("expose", false, "If true, a public, external service is created for the container(s) which are run")
>>> >  	cmd.Flags().String("service-generator", "service/v2", "The name of the generator to use for creating a service.  Only used if --expose is true")
>>> >  	cmd.Flags().String("service-overrides", "", "An inline JSON override for the generated service object. If this is non-empty, it is used to override the generated object. Requires that the object supply a valid apiVersion field.  Only used if --expose is true.")
>>> > +	cmd.Flags().Bool("silent", false, "If true, hide the 'waiting for pod to be running' message")
>>>
>>> What'd other messages to be hidden? Something like when kubectl delete
>>> pod/foo --silent we don't print pod "foo" deleted but only print error
>>> messages?
>>>
>>> Yep, only print error messages. When passing --silent you would rely on
>>> the return code to determine whether command succeeded.
>>>
>>> —
>>> You are receiving this because you were assigned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/kubernetes/kubernetes/pull/28801/files/deb67df071895d8eb9e41a1ac8c3d46c7fcef768#r70349834>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe/ABG_px5dVODgN287dOhxlfU8Z9Ey7BUoks5qUsUwgaJpZM4JJz6B>
>>> .
>>>
>>
>>
>